saleor/plugins/webhook/plugin.py

```diff
@@ -525,6 +525,36 @@ def order_fully_paid(self, order: "Order", previous_value: Any) -> Any:
                 order_data, event_type, webhooks, order, self.requestor
             )
 
+    def order_paid(self, order: "Order", previous_value: Any) -> Any:
+        if not self.active:
+            return previous_value
+        event_type = WebhookEventAsyncType.ORDER_PAID
+        if webhooks := get_webhooks_for_event(event_type):
+            order_data = generate_order_payload(order, self.requestor)
+            trigger_webhooks_async(
+                order_data, event_type, webhooks, order, self.requestor
+            )
+
+    def order_refunded(self, order: "Order", previous_value: Any) -> Any:
+        if not self.active:
+            return previous_value
+        event_type = WebhookEventAsyncType.ORDER_REFUNDED
+        if webhooks := get_webhooks_for_event(event_type):
+            order_data = generate_order_payload(order, self.requestor)
+            trigger_webhooks_async(
+                order_data, event_type, webhooks, order, self.requestor
+            )
+
+    def order_fully_refunded(self, order: "Order", previous_value: Any) -> Any:
+        if not self.active:
+            return previous_value
+        event_type = WebhookEventAsyncType.ORDER_FULLY_REFUNDED
+        if webhooks := get_webhooks_for_event(event_type):
+            order_data = generate_order_payload(order, self.requestor)
+            trigger_webhooks_async(
+                order_data, event_type, webhooks, order, self.requestor
+            )
```

Comment:
Looks like we can create a function from this part that will that the `event_type`. It's the same for each webhook that you added.

Reply:
@IKarbowiak I was considering this approach, but felt hesitant since it would require refactoring the entire plugin file. I didn't want to mix refactoring with adding a new feature, especially since we plan to remove the WebhookPlugin and migrate its logic to Saleor. Given this, would you like me to refactor the entire file in this pull request?

Reply:
Ok then leave it as it is :)